iphone/Classes/ContactsModule.m

```diff
@@ -900,131 +900,116 @@ -(void)peoplePickerNavigationControllerDidCancel:(ABPeoplePickerNavigationContro
-				CFRelease(val);
-			}
-		} else {
-			//birthdays for iOS8 is multivalue and NOT kABPersonBirthdayProperty only in DELEGATE, but undocumented in Apple
-			if ([TiUtils isIOS8OrGreater] && property == appleUndocumentedBirthdayProperty) {
-				CFTypeRef val = nil;
-				if (identifier == 0) {
-					propertyName = @"birthday";
-					val = ABRecordCopyValue(person, kABPersonBirthdayProperty);
-				} else {
-					propertyName = @"alternateBirthday";
-					val = ABRecordCopyValue(person, kABPersonAlternateBirthdayProperty);
-				}
-				if (val != NULL) {
-					value = [[(id)val retain] autorelease];  // Force toll-free bridging & autorelease
-					CFRelease(val);
-				}
-			} else {
-				propertyName = [[[TiContactsPerson multiValueProperties] allKeysForObject:[NSNumber numberWithInt:property]] objectAtIndex:0];
-				ABMultiValueRef multival = ABRecordCopyValue(person, property);
-				CFIndex index = ABMultiValueGetIndexForIdentifier(multival, identifier);
-
-				CFTypeRef val = ABMultiValueCopyValueAtIndex(multival, index);
-				if (val != NULL) {
-					value = [[(id)val retain] autorelease];  // Force toll-free bridging & autorelease
-					CFRelease(val);
-				}
-
-				CFStringRef CFlabel = ABMultiValueCopyLabelAtIndex(multival, index);
-				NSArray *labelKeys = [[TiContactsPerson multiValueLabels] allKeysForObject:(NSString *)CFlabel];
-				if ([labelKeys count] > 0) {
-					label = [NSString stringWithString:[labelKeys objectAtIndex:0]];
-				} else {
-					// Hack for Exchange and other 'cute' setups where there is no label associated with a multival property;
-					// in this case, force it to be the property name.
-					if (CFlabel != NULL) {
-						label = [NSString stringWithString:(NSString *)CFlabel];
-					}
-					// There may also be cases where we get a property from the system that we can't handle, because it's undocumented or not in the map.
-					else if (propertyName != nil) {
-						label = [NSString stringWithString:propertyName];
-					}
-				}
-				if (CFlabel != NULL) {
-					CFRelease(CFlabel);
-				}
-				CFRelease(multival);
-			}
-		}
-
-		NSDictionary *dict = [NSDictionary dictionaryWithObjectsAndKeys:personObject, @"person", propertyName, @"property", value, @"value", label, @"label", nil];
-		[self _fireEventToListener:@"selectedProperty" withObject:dict listener:selectedPropertyCallback thisObject:nil];
-		[[TiApp app] hideModalController:picker animated:animated];
-		return NO;
-	}
-	return YES;
-}
 // Called after a person has been selected by the user. New in iOS 8
 - (void)peoplePickerNavigationController:(ABPeoplePickerNavigationController*)peoplePicker didSelectPerson:(ABRecordRef)person
 {
-    [self peoplePickerNavigationController:peoplePicker shouldContinueAfterSelectingPerson:person];
+    if (selectedPersonCallback) {
+        TiContactsPerson* personObject = nil;
+        if ([TiUtils isIOS8OrGreater] && (ABAddressBookGetAuthorizationStatus() != kABAuthorizationStatusAuthorized)) {
```

Comment:
`[TiUtils isIOS8OrGreater]` should not be required when a) the delegate only fires for iOS8+ and b) we target iOS 8+.